Source/EventFlow.MsSql/ReliablePublish/MsSqlReliablePublishPersistence.cs

```diff
@@ -0,0 +1,132 @@
+﻿// The MIT License (MIT)
+//
+// Copyright (c) 2015-2018 Rasmus Mikkelsen
+// Copyright (c) 2015-2018 eBay Software Foundation
+// https://github.com/eventflow/EventFlow
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of
+// this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to
+// use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EventFlow.Aggregates;
+using EventFlow.Core;
+using EventFlow.EventStores;
+using EventFlow.PublishRecovery;
+using EventFlow.Subscribers;
+
+namespace EventFlow.MsSql.ReliablePublish
+{
+    public sealed class MsSqlReliablePublishPersistence : IReliablePublishPersistence
+    {
+        private readonly IMsSqlConnection _msSqlConnection;
+
+        public MsSqlReliablePublishPersistence(IMsSqlConnection msSqlConnection)
+        {
+            _msSqlConnection = msSqlConnection;
+        }
+
+        public async Task MarkPublishedAsync(IIdentity aggregateIdentity, IReadOnlyCollection<IDomainEvent> domainEvents)
```

Comment:
I think that by taking aggregateIdentity parameter you are leaking your implementation into interface. 
This could be determined from domainEvents and some other implementations could be fine with marking events from different aggregateIdentity in one roundtrip.

Reply:
However, I don't think it will be easy to completely remove this from interface.

Reply:
It's some kind of 'guarantee' that all domain events should related with one aggregate...

Reply:
Why we need this kind of guarantee?